src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,39 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMUM_NUMBER_OF_CHARACTERS = 6;
+    private static final int MINIMUM_AGE_FOR_REGISTRATION = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User must be initialized");
+        }
+        if (user.getLogin() == null) {
+            throw new RegistrationException("The login must be initialized");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("Such a login already exists");
+        }
```

Comment:
validation if user exists should be performed when all input parameters are correct